### PR TITLE
Move from labeling contents of /var/run to /run

### LINKED ISSUE #242: `/run/motd` and `/run/motd.d` paths not getting context `etc_t`
The changes in #230 and #232 gave `/run/motd` and `/run/motd.d(/.*)?` paths the context `etc_t`, which allows `sshd` to read these paths and display MOTD messages here.

Trying out the MOTD functionality (in a Fedora 29 cloud base Vagrant box), the paths still have the context `var_run_t` despite the setting to change the context to `etc_t` being present.

```
# dnf upgrade -y
...
# rpm -qa | grep "selinux-policy"
selinux-policy-targeted-3.14.2-46.fc29.noarch
selinux-policy-3.14.2-46.fc29.noarch
# ls -Z /run | grep motd
               system_u:object_r:var_run_t:s0 motd.d
# semanage fcontext -l | grep motd
...
/run/motd                                          regular file       system_u:object_r:etc_t:s0 
/run/motd.d(/.*)?                                  all files          system_u:object_r:etc_t:s0 
...
```

Doing the following gives the `/run/motd.d` path the `etc_t` context as desired, and the rule appears in the file contexts:

```
# semanage fcontext -a -t etc_t "/var/run/motd.d(/.*)?"
# restorecon -Rnv /run
Would relabel /run/motd.d from system_u:object_r:var_run_t:s0 to system_u:object_r:etc_t:s0
...
# semanage fcontext -l | grep motd
...
/run/motd                                          regular file       system_u:object_r:etc_t:s0 
/run/motd.d(/.*)?                                  all files          system_u:object_r:etc_t:s0 
...
/var/run/motd.d(/.*)?                              all files          system_u:object_r:etc_t:s0 
```

If adding a rule with the path `/run/motd.d(/.*)?`, the following error is given:

```
# semanage fcontext -a -t etc_t "/run/motd.d(/.*)?"
ValueError: File spec /run/motd.d(/.*)? conflicts with equivalency rule '/run /var/run'; Try adding '/var/run/motd.d(/.*)?' instead
```

It seems that adding the context for a path under `/run` is causing a conflict, and is why the change to give `/run/motd` and `/run/motd.d` context `etc_t` is not being propagated. I haven't verified this by building the policy with the changed settings on my system yet, though.

Would the best convention here be to change the paths in `policy/modules/kernel/files.fc` to `/var/run/motd` and `/var/run/motd.d(/.*)?`, in order to successfully label `/run/motd` and `/run/motd.d(/.*)?` with `etc_t`?

Thanks!

cc @dustymabe, @lucab, @LorbusChris

### PR DESCRIPTION
**Pull request by LorbusChris**
This commit switches labelling from /var/run to /run.
The equivalence rule for /var/run and /run is rearranged to
better reflect the reality /var/run being a symlink to /run.

### CODE REVIEW

**Comment by LorbusChris** [comment]
@rhatdan @wrabcak I've resurrected this. Right now, all the labeling in `/var/run`, which is a symlink to `/run`, makes the policy confusing. How can we move forward with this? What else would it need to deprecate `/var/run`?

**Comment by rhatdan** [comment]
I have no problem with moving it. it was originally there because most distributions still used /var/run.  But about 8-10 years ago the defaults changed to /run at least for Fedora and eventually RHEL.   There was an equivalency rule that said label /run like it was in /var/run which made sure /run was labelled correctly.  Switching to /run and then adding an equivalency rule for /var/run would make the policy better match reality.

**Comment by LorbusChris** [comment]
@rhatdan I've updated `config/file_contexts.subs_dist` with the equivalency rules. Please let me know if there is anything else needed here!

**Comment by LorbusChris** [comment]
At a quick glance, I'm not sure where the CI syntax error is coming from. It wasn't there before and the PR does not touch the mentioned line?!

**Comment by wrabcak** [comment]
Hi @LorbusChris , 
I need to fix the CI, it's issue in policy sources unrelated to this PR, then I merge this PR and will see what happen. It looks like a quite dangerous change.

**Comment by wrabcak** [comment]
@LorbusChris , 
CI Fixed. 
But we need to change this also in selinux-policy-contrib repo. Could you create same PR also against selinux-policy-contrib repo? 

Thanks,
Lukas.

File: policy/modules/contrib/apache.fc [hunk 1]
```diff
@@ -218,5 +218,6 @@ ifdef(`distro_debian', `
 /var/www/svn/conf(/.*)?			gen_context(system_u:object_r:httpd_sys_content_t,s0)
 
 /var/log/dirsrv/admin-serv(/.*)?	gen_context(system_u:object_r:httpd_log_t,s0)
-/var/run/dirsrv/admin-serv.*	gen_context(system_u:object_r:httpd_var_run_t,s0)
+/run/dirsrv/admin-serv.*	gen_context(system_u:object_r:httpd_var_run_t,s0)
 /opt/dirsrv/var/run/dirsrv/dsgw/cookies(/.*)?       gen_context(system_u:object_r:httpd_var_run_t,s0)
+#/opt/dirsrv/run/dirsrv/dsgw/cookies(/.*)?       gen_context(system_u:object_r:httpd_var_run_t,s0)
```
Comment (LorbusChris) [inline policy/modules/contrib/apache.fc:223]: NTS

File: policy/modules/contrib/dbus.fc [hunk 2]
```diff
@@ -26,10 +26,11 @@ ifdef(`distro_gentoo',`
...
 
-/var/run/dbus(/.*)?		gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
+/run/dbus(/.*)?		gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
 
-/var/run/user/[0-9]+/dbus(/.*)? 	gen_context(system_u:object_r:session_dbusd_tmp_t,s0)
+/run/user/[0-9]+/dbus(/.*)? 	gen_context(system_u:object_r:session_dbusd_tmp_t,s0)
 
 ifdef(`distro_redhat',`
 /var/named/chroot/var/run/dbus(/.*)?	gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
+#/var/named/chroot/run/dbus(/.*)?	gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
```
Comment (LorbusChris) [inline policy/modules/contrib/dbus.fc:35]: NTS

**Review by LorbusChris — COMMENTED** [review]
Rebased (sorry for resurrecting this...).

There are two paths that might make sense to add with this change, I've added the comments below as markers

**Comment by LorbusChris** [comment]
@wrabcak I know this is going to be a pain to review, but please take a look when you get the chance

**Comment by wrabcak** [comment]
@zpytela, Please take a look. 

Thanks,
Lukas.

**Comment by LorbusChris** [comment]
I still think this would be a nice to have, but I don't have the time to look after this PR.

**Pull request closed without merging**